### PR TITLE
Replace Block.getChildren with ColumnarRow

### PR DESCRIPTION
**Pull request by homar**
## Description
We should not use Block.getChildren(). It is internal routine not intended to use directly. May not work always.
> Is this change a fix, improvement, new feature, refactoring, or other?
a fix
> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
a connector
> How would you describe this change to a non-technical end user or system administrator?
technical change, not interesting for the end user
## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java [hunk 1]
```diff
@@ -400,8 +402,9 @@ public void updateRows(Page page, List<Integer> columnValueAndRowIdChannels)
                 updatedColumnIndex++;
             }
             else {
-                Block unmodifiedColumns = rowIdBlock.getChildren().get(1);
-                fullPage[targetChannel] = unmodifiedColumns.getChildren().get(unmodifiedColumnIndex);
+                ColumnarRow columnarRow = toColumnarRow(rowIdBlock);
```
Comment (alexjo2144) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java:405]: Put this outside of the for loop

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java [hunk 2]
```diff
@@ -393,15 +395,16 @@ public void updateRows(Page page, List<Integer> columnValueAndRowIdChannels)
...
         for (int targetChannel = 0; targetChannel < allDataColumns.size(); targetChannel++) {
             if (updatedColumns.contains(allDataColumns.get(targetChannel))) {
                 fullPage[targetChannel] = page.getBlock(columnChannelMapping.get(updatedColumnIndex));
                 updatedColumnIndex++;
             }
             else {
-                Block unmodifiedColumns = rowIdBlock.getChildren().get(1);
-                fullPage[targetChannel] = unmodifiedColumns.getChildren().get(unmodifiedColumnIndex);
+                Block unmodifiedColumns = columnarRow.getField(1);
+                fullPage[targetChannel] = toColumnarRow(unmodifiedColumns).getField(unmodifiedColumnIndex);
```
Comment (alexjo2144) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java:407]: You can move this one out as well

Comment (homar) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java:407]: I actually tried that and it made some tests fail with IndexOutOfBound in `Block unmodifiedColumns = columnarRow.getField(1);`

Comment (alexjo2144) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdatablePageSource.java:407]: Right, that might happen if all rows are being updated. You could make the field Optional or nullable. Might be worth it if `toColumnarRow` might do a meaningful amount of work

**Comment by homar** [comment]
@alexjo2144 is it ok now ?

**Review by alexjo2144 — APPROVED** [review]

**Comment by homar** [comment]
@findepi can we merge this ?

**Pull request merged**